### PR TITLE
fix bugs in the gridded runoff capabilities

### PR DESCRIPTION
**Pull request by martynpclark**

### CODE REVIEW

File: route/build/src/remap.f90 [hunk 1]
```diff
@@ -151,6 +163,8 @@ subroutine remap_2D_runoff(runoff_data, remap_data, structHRU2seg, basinRunoff,
 
   end do   ! looping through basins in the routing layer
```
Comment (nmizukami) [inline route/build/src/remap.f90]: This comment and the line 288 should be 
> looping through basins in the mapping layer

File: route/build/src/remap.f90 [hunk 2]
```diff
@@ -113,6 +113,18 @@ subroutine remap_2D_runoff(runoff_data, remap_data, structHRU2seg, basinRunoff,
     jj = remap_data%j_index(ixOverlap)
     ii = remap_data%i_index(ixOverlap)
 
+    ! check i-indices
```
Comment (nmizukami) [inline route/build/src/remap.f90]: This is good catch.  Maybe more descriptive comments help?  
something like...

> 'When computing runoff at ', jHRU, 'th HRU, No valid runoff value is found at i index =', ii, 'in runoff data.'

File: route/build/src/remap.f90 (hunk 1 above)
Comment (martynpclark) [reply route/build/src/remap.f90]: Thanks @nmizukami -- can you implement this change?

**Comment by martynpclark** [comment]
Thanks @nmizukami -- can you implement these changes?

**Pull request merged**